modules/spring-config/src/main/java/org/mule/runtime/config/spring/dsl/processor/ConfigLineProvider.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.config.spring.dsl.processor;
+
+/**
+ * A configuration line provider allows to have a parent child relationship
+ * between {@code ConfigLine} while keeping the object immutable.
+ *
+ * @since 4.0
+ */
+public interface ConfigLineProvider
```

Comment:
I will leave the interface just because it clarifies a little bit better why I need to use it.